### DISCUSSION TITLE
The case for HealthChecks in V8

### DISCUSSION THREAD

**Discussion by slacki123** [dotnet/dotnet-operator-sdk#716] [Q&A] [answered]
Hi, 

I noticed that the health/readiness check setup that existed in v7, no longer exists in v8. Has it been completely removed or is there some different way to configure them?

If they were removed, then was there a reason for it?

I have no problem setting it up manually, but it adds quite a few of code to maintain, probably would be nicer to have it configurable out of the box.

**Comment by buehler** [comment]
Hey @slacki123 
They were not completely removed, just the passed through reference to the original function of ASP.NET. Since the operator is now split up such that one can make an operator without the whole ASP.NET overhead, you need to use the "normal" operator and the web reference of .NET.

If you don't need webhook support, you can skip the web operator package. However, to provide these checks, you should just use the default ones of ASP.NET.

**Reply by slacki123** [reply]
Forgive me if I misunderstand 

I was referring to these healthcheck configurations:
https://github.com/buehler/dotnet-operator-sdk/blob/maintenance/7.x/src/KubeOps/Operator/ApplicationBuilderExtensions.cs#L27

which I now configure in my own Program.cs file rather than it coming out of the box as in v7 `ApplicationBuilderExtensions`.

I understand that the ASP.NET provides the linked methods for configuration, but I was more referring to whether they should be configured in KubeOps or by the user implementing it?

**Reply by buehler** [reply, marked as answer]
Ah, alright. The main point of v8 was to separate operator logic and ASP.NET logic. The ASP part produced a big overhead that was not needed if you don't use webhooks. As a side effect, this removed the possibility of healthchecks as well.

The intension is to configure the healthchecks by yourself if you need them.

It does not really matter (in my opinion at least), if you write `builder.AddHealthCheck()` with the operator builder or if you write `services.AddHealthCheck` on the configurable services in asp.net.

So since the healthchecks have no direct connection to the operator SDK per se, I think the user that implements the operator should decide if and when they want to register the checks.

This allows other developers to just ignore the checks altogether and allows of a better separation of concerns.

**Reply by slacki123** [reply]
I see, that's fair enough. I just wanted to know the way to approach this. Thanks :)
